nemo-physical/src/columnar/operations/arithmetic/expression.rs

```diff
@@ -0,0 +1,471 @@
+//! Implements the functionality to represent and evaluate complex arithmetic expressions
+
+use std::fmt::{Debug, Display};
+
+use super::traits::ArithmeticOperations;
+
+/// Tree representing an arithmetic expression
+#[derive(Clone)]
+pub enum ArithmeticTree<T> {
```

Comment:
I'd imagine, it would be helpful for this enum to be more structural e.g.
```rust
enum ArithmeticTree<> {
  Constant(..),
  Reference(..),
  BinaryOperation(..),
  UnaryOperation(..)
}
```

Probably postpone it though, since this tree structure should live in the nemo module, not in the physical layer.

Reply:
Yeah, I mean this will get replaced anyways. But good idea nonetheless.